docs/csharp/versioning.md

```diff
@@ -0,0 +1,132 @@
+a way that information that is likely to change regularly is stored in the `app.config` file, this way when such information is updated
+the config file of older versions just needs to be replaced with the new one without the need for recompilation of the library.
+
+## Consuming Libraries
+
+As a developer that consumes .NET libraries built by other developers you're most likely aware that a new version of a library might not be fully compatible with your project
+and you might often find yourself having to update your code to work with those changes.
+
+Lucky for you C# and the .NET ecosystem comes with features and techniques that allow us to easily update our app to work with new versions of libraries that might introduce breaking changes.
+
+### Assembly Binding Redirection
+
+You can use the `app.config` file to update the version of a library your app uses. By adding what is called a [*binding redirect*](https://msdn.microsoft.com/en-us/library/7wd6ex19(v=vs.110).aspx) your
+can use the new library version without having to recompile your app. The following example shows how you would update
+your app's `app.config` file to use the `1.0.1` patch version of `ReferencedLibrary` instead of the `1.0.0` version it was originally compiled with.
+
+```xml
+<dependentAssembly>
+    <assemblyIdentity name="ReferencedLibrary" publicKeyToken="32ab4ba45e0a69a1" culture="en-us" />
+    <bindingRedirect oldVersion="1.0.0" newVersion="1.0.1" />
+</dependentAssembly>
+```
+
+> [!NOTE]
+> This approach will only work if the new version of `ReferencedLibrary` is binary compatible with your app.
+> See the [Backwards Compatibility](#backwards-compatibility) section above for changes to look out for when determining compatibility.
+
+### new
+
+You use the `new` modifier to hide inherited members of a base class.
+This makes your derived classes immune to changes made to the inherited members of the base class
+
+Take the following example:
+
+[!code-csharp[Sample usage of the 'new' modifier](../../samples/csharp/versioning/new/Program.cs#sample)]
+
+**Output**
+
+```
+A base method
+A derived method
+```
+
+From the example above you can see how `DerivedClass` hides the `MyMethod` method present in `BaseClass`.
+This means that when a base class in the new version of a library adds a member that already exists in your derived class, you can
+simply use the `new` modifier on your derived class member to hide the base class member.
+
+When no `new` modifier is specified, a derived class will by default hide conflicting members in a base class,
+although a compiler warning will be generated the code will still compile. This means that simply adding new members to an existing class
+makes that new version of your library both source and binary compatible with code that depends on it.
+
+### override
+
+The `override` modifier is the opposite of the `new` modifier, it extends the implementation of a base class member rather than
```

Comment:
I wouldn't say that override is the opposite of new. I'd just say:

"The `override` modifier means a derived implementation extends the implementation of a base class member..."